### PR TITLE
[FIX/ADD] transfer_dimension with domain, budget_control_transfer_dimension_constraint

### PR DESCRIPTION
**Pull request by Saran440**
- Fix filter budget_control_transfer_dimension can domain with sequence
- Add new module budget_control_transfer_dimension_constraint for constraint transfer with dimension

### CODE REVIEW

File: budget_control_transfer_dimension/models/budget_transfer_item.py [hunk 1]
```diff
@@ -55,7 +159,9 @@ def _compute_source_analytic_tag_all(self):
                 )
             ):
                 continue
-            doc.source_analytic_tag_ids = analytic_tag_ids
+            doc.source_analytic_tag_ids = (
+                len(analytic_tag_ids) == 1 and analytic_tag_ids or False
+            )
```
Comment (ps-tubtim) [inline budget_control_transfer_dimension/models/budget_transfer_item.py]:
```suggestion
            doc.source_analytic_tag_ids = (
                len(analytic_tag_ids) == len(dimension_fields) and analytic_tag_ids or False
            )
```

File: budget_control_transfer_dimension/models/budget_transfer_item.py [hunk 2]
```diff
@@ -72,25 +178,41 @@ def _compute_target_analytic_tag_all(self):
                 )
             ):
                 continue
-            doc.target_analytic_tag_ids = analytic_tag_ids
+            doc.target_analytic_tag_ids = (
+                len(analytic_tag_ids) == 1 and analytic_tag_ids or False
+            )
```
Comment (ps-tubtim) [inline budget_control_transfer_dimension/models/budget_transfer_item.py]:
```suggestion
            doc.target_analytic_tag_ids = (
                len(analytic_tag_ids) == len(dimension_fields) and analytic_tag_ids or False
            )
```

File: budget_control_transfer_dimension/models/budget_transfer_item.py (hunk 1 above)
Comment (ps-tubtim) [reply budget_control_transfer_dimension/models/budget_transfer_item.py]: The dimensions must more than 1.

Comment (Saran440) [reply budget_control_transfer_dimension/models/budget_transfer_item.py]: user should select 1 only.
if analytic_tag_ids has 1 line, This function will help user default dimension.
if analytic_tag_ids more than 1 user must select field manual.

**Pull request merged**